tests/test_nominal_column.py

```diff
@@ -136,4 +136,20 @@ def test_changing_copy(self):
 
     def test_metadata(self):
         """ Ensure metadata is copied correctly or deep_copy """
-        assert self.copy.metadata == self.orig.metadata, 'Copied metadata should be equivilent'
+        assert self.copy.metadata == self.orig.metadata, 'Copied metadata should be equivalent'
+
+
+class TestBugFixes(TestCase):
+    """ Specific tests for bug fixes """
+    def test_comparison_of_different_object_types(self):
+        """
+        Fix bug whereby floats were being passed into NominalColumn
+        from `self.observed = NominalColumn(arr)` but as `dtype=object`
+        resulting in `TypeError: unorderable types: int() > str()` in
+        python 3.x only
+        """
+        input_list = [100, 'c', 13, 15, np.nan, np.nan]
+        object_arr = np.array(input_list, dtype=object)
+        vector = CHAID.NominalColumn(object_arr)
+
+        assert [vector.metadata[x] for x in vector.arr] == ['<missing>' if x != x else x for x in input_list]
```

Comment:
@xulaus not the usual test, but the order of `set()` is unpredictable, so couldn't proceed with historic tests. I also couldn't change the type and then run `np.unique` because it would then fail to resub them further down. This is the only way I could think handling that case, otherwise we have to ask the user to convert the column before running it through the CHAID, but only in python3. Argh this issue is annoying.